hydrolib/core/utils.py

```diff
@@ -69,3 +72,55 @@ def get_substring_between(source: str, start: str, end: str) -> Optional[str]:
         return None
 
     return source[index_start:index_end]
+
+
+class HVersion(SemVer):
+    @classmethod
+    def coerce(cls, *args, **kwargs):
+        version = super().coerce(*args, **kwargs)
+        return cls(**version.__dict__)
+
+
+class FMVersion(HVersion):
```

Comment:
I don't quite like this FM-related class here, inside a "neutral" utils namespace.
Also it suggest that this is "the" FM version, whereas it is only the numbering schema, so consider renaming this class to: `class FMStyleVersion`, or `DHydroStyleVersion`, or `ZeroPrefixedVersion`.